packages/publish/src/lib/pack-directory.ts

```diff
@@ -69,6 +73,47 @@ export async function packDirectory(_pkg: Package, dir: string, options: PackCon
   );
 }
 
+export const PUBLISH_CONFIG_FIELDS = [
+  'bin',
+  'main',
+  'exports',
+  'types',
+  'typings',
+  'module',
+  'browser',
+  'esnext',
+  'es2015',
+  'unpkg',
+  'umd:main',
+  'cpu',
+  'os',
+];
+
+async function overwritePublishConfig(manifestLocation: string) {
+  const manifest = await fs.readJSON(manifestLocation);
+
+  if (manifest.publishConfig === undefined) {
+    return;
+  }
+
+  const publishConfig = { ...manifest.publishConfig };
+
+  PUBLISH_CONFIG_FIELDS.forEach((key) => {
+    if (key in publishConfig) {
+      manifest[key] = publishConfig[key];
+      delete publishConfig[key];
```

Comment:
I didn't test the code but I wonder what's the reason that you have to delete the object property? I see you're also deleting a few lines below as well